x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTests.java

```diff
@@ -869,6 +870,79 @@ public void testMaybeRewriteRealmRef() {
+        final Map<String, Object> metadata = Map.ofEntries(
+            Map.entry(AuthenticationField.API_KEY_ID_KEY, apiKeyId),
+            Map.entry(AuthenticationField.API_KEY_NAME_KEY, apiKeyName),
+            Map.entry(AuthenticationField.API_KEY_ROLE_DESCRIPTORS_KEY, new BytesArray("""
+                {"base_role":{"cluster":["all"],
+                "remote_indices":{"names":["logs-*"],"privileges":["read"],"clusters":["my_cluster*","other_cluster"]}}
+                }""")),
+            Map.entry(AuthenticationField.API_KEY_LIMITED_ROLE_DESCRIPTORS_KEY, new BytesArray("""
+                {"limited_by_role":{"cluster":["*"],
+                "remote_indices":{"names":["logs-*-*"],"privileges":["write"],"clusters":["my_cluster*"]}}
+                }"""))
+        );
+
+        final Authentication original = AuthenticationTestHelper.builder()
+            .apiKey()
+            .metadata(metadata)
+            .transportVersion(Authentication.VERSION_API_KEYS_WITH_REMOTE_INDICES)
+            .build();
+
+        // pick a version before that of the authentication instance to force a rewrite
+        final TransportVersion olderVersion = TransportVersionUtils.randomVersionBetween(
+            random(),
+            Authentication.VERSION_API_KEY_ROLES_AS_BYTES,
+            original.getEffectiveSubject().getTransportVersion()
+        );
+
+        final Map<String, Object> rewrittenMetadata = original.maybeRewriteForOlderVersion(olderVersion)
+            .getEffectiveSubject()
+            .getMetadata();
+        assertThat(rewrittenMetadata.keySet(), equalTo(original.getAuthenticatingSubject().getMetadata().keySet()));
+        assertThat(
+            ((BytesReference) rewrittenMetadata.get(AuthenticationField.API_KEY_ROLE_DESCRIPTORS_KEY)).toBytesRef(),
+            equalTo(new BytesArray("""
+                {"base_role":{"cluster":["all"]}}""").toBytesRef())
+        );
+        assertThat(
+            ((BytesReference) rewrittenMetadata.get(AuthenticationField.API_KEY_LIMITED_ROLE_DESCRIPTORS_KEY)).toBytesRef(),
+            equalTo(new BytesArray("""
+                {"limited_by_role":{"cluster":["*"]}}""").toBytesRef())
+        );
+    }
+
+    public void testMaybeRemoveRemoteIndicesFromRoleDescriptors() {
+        final BytesReference roleWithoutRemoteIndices = new BytesArray("""
+            {"user_role":{"cluster":["all"]}}""");
+
+        // role without remote indices should stay the same
+        assertThat(
+            roleWithoutRemoteIndices.toBytesRef(),
+            equalTo(Authentication.maybeRemoveRemoteIndicesFromRoleDescriptors(roleWithoutRemoteIndices).toBytesRef())
+        );
+
+        // role with remote indices should be filtered
+        assertThat(
+            roleWithoutRemoteIndices.toBytesRef(),
+            equalTo(Authentication.maybeRemoveRemoteIndicesFromRoleDescriptors(new BytesArray("""
+                {"user_role":{"cluster":["all"],
```

Comment:
Optional: might randomly omit the cluster priv to ensure that we get a well-formed, empty role descriptor, after removing.